plugins/modules/packaging/os/zypper_repository.py

```diff
@@ -382,12 +385,54 @@ def exit_unchanged():
         if not alias and state == "present":
             module.fail_json(msg='Name required when adding non-repo files.')
 
+    # Download and parse .repo file to ensure idempotency
+    if repo and repo.endswith('.repo'):
+        response, info = fetch_url(module=module, url=repo, force=True)
+        if response and info['status'] == 200:
+            repofile = configparser.ConfigParser()
```

Comment:
As mentioned yesterday on IRC, I think the code would be cleaner if you do the fail in the primary `if` branch. Then you don't have to indent all the code which is inside it right now (same for the other two `if`s where `else` basically only calls `module.fail_json`):
```suggestion
        if not response or info['status'] != 200:
            module.fail_json(msg='Error downloading .repo file from provided URL')
        repofile = configparser.ConfigParser()
```

Reply:
Thank you very much, indeed, way better without the else statements and with reduced indentation, changes pushed.